lib/Jsrt/Jsrt.cpp

```diff
@@ -1215,12 +1215,22 @@ CHAKRA_API JsStringToPointer(_In_ JsValueRef stringValue, _Outptr_result_buffer_
 
 CHAKRA_API JsConvertValueToString(_In_ JsValueRef value, _Out_ JsValueRef *result)
 {
+    PARAM_NOT_NULL(result);
+    *result = nullptr;
+
+    if (value != nullptr && Js::JavascriptString::Is(value))
+    {
+        return ContextAPINoScriptWrapper_NoRecord([&](Js::ScriptContext *scriptContext) -> JsErrorCode {
+            VALIDATE_INCOMING_REFERENCE(value, scriptContext);
+            *result = Js::CrossSite::MarshalVar(scriptContext, value,
```

Comment:
I believe that this isn't required after the `VALIDATE_INCOMING_REFERENCE`, it does some marshaling on the `value` parameter already. We should just be able to do `*result = value` here.

Reply:
Well, actually `VALIDATE_INCOMING_REFERENCE` totally unnecessary here.

Reply:
It enters here only when `Js::JavascriptString::Is` `true`

Reply:
`VALIDATE_INCOMING_REFERENCE` also does some other checks too though; I'm not sure if they are strictly required here, but it checked that it was the same runtime as well I think. Are you sure that it isn't necessary here?

Reply:
```
#define VALIDATE_INCOMING_OBJECT(p, scriptContext) \
        { \
            VALIDATE_JSREF(p); \
            if (!Js::JavascriptOperators::IsObject(p)) \
            { \
                return JsErrorArgumentNotObject; \
            } \
            MARSHAL_OBJECT(p, scriptContext)   \
        }
```

```
#define VALIDATE_JSREF(p) \
    if (p == JS_INVALID_REFERENCE) \
    { \
        return JsErrorInvalidArgument; \
    } \
```

>  it checked that it was the same runtime as well I think.

?

Well, as I mentioned, we do `!nullptr && Js::JavascriptString::Is` check.. and Marshal but we don't do `JS_INVALID_REFERENCE` check.

```
#ifdef __cplusplus
    const JsRef JS_INVALID_REFERENCE = 0;
#else
    #define JS_INVALID_REFERENCE (JsRef)0
#endif
```

Reply:
```cpp
#define MARSHAL_OBJECT(p, scriptContext) \
        Js::RecyclableObject* __obj = Js::RecyclableObject::FromVar(p); \
        if (__obj->GetScriptContext() != scriptContext) \
        { \
            if(__obj->GetScriptContext()->GetThreadContext() != scriptContext->GetThreadContext()) \
            {  \
                return JsErrorWrongRuntime;  \
            }  \
            p = Js::CrossSite::MarshalVar(scriptContext, __obj); \
}
```

That's the part that I'm referring to; the comparison of the thread context. I'm not familiar with what the consequences of ignoring that are.

Reply:
Since we know it will pass `recyclableObject::FromVar` straight, this one looks okay too.